docker/Dockerfile_txsim

```diff
@@ -0,0 +1,49 @@
+# stage 1 Generate celestia-appd Binary
+FROM docker.io/golang:1.20.5-alpine3.17 as builder
```

Comment:
```suggestion
FROM docker.io/golang:1.20.6-alpine3.17 as builder
```
same as in [celestia-app](https://github.com/celestiaorg/celestia-app/blob/main/Dockerfile#L2)

Reply:
Ack. Changed to 1.20.6